orttraining/orttraining/python/experimental/checkpoint.py

```diff
@@ -11,11 +11,11 @@
 ################################################################################
 
 
-def experimental_state_dict(ort_trainer):
+def experimental_state_dict(ort_trainer, include_optimizer_state=True):
     if not ort_trainer._training_session:
         warnings.warn("ONNX Runtime training session is not initialized yet. "
                         "Please run train_step or eval_step at least once before calling state_dict().")
-        return {}
+        return ort_trainer._state_dict
```

Comment:
what is the motivation for this change? is it possible that `_state_dict` is available when `_training_session` is `None`? In that case should we update the warning above

Reply:
That is actually a good point. The motivation was that it is possible the user calls `load_state_dict(my_state)` and before calling `train_step(input)` or `eval_step(nput)`  - which would create a training session and effectively load the state_dict into the model - they call `state_dict()` just to check whether the state was properly preloaded into the class.

It would be strange that the user laods a state into ORTTrainer and then doesnt have a way to find out whther the state was "processed" or at least "stashed"

By making this change, I don't expect any existing feature/model to break, as nothing really happens before training.

Reply:
I think the warning is still necessary, as the state being returned is not complete and the user really needs to call train_step or eval_step. What would be your suggestion for rephrasing it?